sql/src/test/java/ddlog/RuntimeTest.java

```diff
@@ -0,0 +1,1064 @@
+                "END");
+        s = node.toString();
+        Assert.assertEquals("if (64'sd0 == 64'sd1){\n" +
+                "\"One\"} else {\n" +
+                "if (64'sd0 == 64'sd2){\n" +
+                "\"Two\"} else {\n" +
+                "\"Other\"}}", s);
+    }
+
+    @Test
+    public void testWhen() {
+        String query = "create view v1 as SELECT DISTINCT CASE WHEN column1 = 1 THEN 1 WHEN 1 < column1 THEN 2 ELSE 3 END FROM t1";
+        String program =
+                this.header(false) +
+                        "typedef Ttmp0 = Ttmp0{col1:signed<64>}\n" +
+                        this.relations(false) +
+                        "relation Rtmp0[Ttmp0]\n" +
+                        "output relation Rv1[Ttmp0]\n" +
+                        "Rv1[v3] :- Rt1[v0],var v2 = Ttmp0{.col1 = if (v0.column1 == 64'sd1){\n" +
+                        "64'sd1} else {\n" +
+                        "if (64'sd1 < v0.column1){\n" +
+                        "64'sd2} else {\n" +
+                        "64'sd3}}},var v3 = v2.";
+        testTranslation(query, program);
+    }
+
+    @Test
+    public void testWhenWNull() {
+        String query = "create view v1 as SELECT DISTINCT CASE WHEN column1 = 1 THEN 1 WHEN 1 < column1 THEN 2 ELSE 3 END FROM t1";
+        String program =
+            this.header(true) +
+                "typedef Ttmp0 = Ttmp0{col1:signed<64>}\n" +
+                this.relations(true) +
+                "relation Rtmp0[Ttmp0]\n" +
+                "output relation Rv1[Ttmp0]\n" +
+                "Rv1[v3] :- Rt1[v0],var v2 = Ttmp0{.col1 = if unwrapBool(a_eq_NR(v0.column1, 64'sd1)){\n" +
+                "64'sd1} else {\n" +
+                "if unwrapBool(a_lt_RN(64'sd1, v0.column1)){\n" +
+                "64'sd2} else {\n" +
+                "64'sd3}}},var v3 = v2.";
+        testTranslation(query, program, true);
+    }
+
+    @Test
+    public void testAlias() {
+        String query = "create view v1 as SELECT DISTINCT t2.column1 FROM t1 AS t2";
+        String program = this.header(false) +
+                "typedef Ttmp0 = Ttmp0{column1:signed<64>}\n" +
+                this.relations(false) +
+                "relation Rtmp0[Ttmp0]\n" +
+                "output relation Rv1[Ttmp0]\n" +
+                "Rv1[v2] :- Rt1[v0],var v1 = Ttmp0{.column1 = v0.column1},var v2 = v1.";
+        testTranslation(query, program);
+    }
+
+    @Test
+    public void testAliasWNull() {
+        String query = "create view v1 as SELECT DISTINCT t2.column1 FROM t1 AS t2";
+        String program = this.header(true) +
+            "typedef Ttmp0 = Ttmp0{column1:Option<signed<64>>}\n" +
+            this.relations(true) +
+            "relation Rtmp0[Ttmp0]\n" +
+            "output relation Rv1[Ttmp0]\n" +
+            "Rv1[v2] :- Rt1[v0],var v1 = Ttmp0{.column1 = v0.column1},var v2 = v1.";
+        testTranslation(query, program, true);
+    }
+
+    @Test
+    public void testScope() {
```

Comment:
This is the same output program as testAlias(). What's being tested here?

Reply:
We should support as much as possible from the SQL syntax.